lib/gitWorkflow.js

```diff
@@ -224,39 +299,52 @@ export class GitWorkflow {
       // - git stash can't infer RD or MD states correctly, and will lose the deletion
       this.deletedFiles = await this.getDeletedFiles()
 
+      // Generate a unique identifier to include in the stash message
+      ctx.stashIdentifier = new Date().toISOString()
```

Comment:
Sort of a breaking change, because `create` doesn't work with `--include-untracked`.

Reply:
That seems worse to me.  It would not be obvious to users why the contents of a stash might change the format of the stash message.  Consistency is better - if `git stash` were consistent, we wouldn't even be in this situation.

Reply:
Sorry, I will have to sped time thinking about this, so won't be merged quickly. To me this feels like a good change, but also one that should be achievable with less changes required.